cmd/nitro/init_test.go

```diff
@@ -554,3 +558,152 @@ func TestOpenInitializeChainDbEmptyInit(t *testing.T) {
+			archiveFiles: []string{
+				"arbitrumdata/000001.ldb",
+				"l2chaindata/000001.ldb",
+				"nodes/000001.ldb",
+				"wasm/000001.ldb",
+			},
+			wantFiles: []string{
+				"arbitrumdata/000001.ldb",
+				"l2chaindata/000001.ldb",
+				"nodes/000001.ldb",
+			},
+		},
+		{
+			name:       "extractAllButWasmWithPrefixDot",
+			importWasm: false,
+			archiveFiles: []string{
+				"./arbitrumdata/000001.ldb",
+				"./l2chaindata/000001.ldb",
+				"./nodes/000001.ldb",
+				"./wasm/000001.ldb",
+			},
+			wantFiles: []string{
+				"arbitrumdata/000001.ldb",
+				"l2chaindata/000001.ldb",
+				"nodes/000001.ldb",
+			},
+		},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			// Create archive with dummy files
+			archiveDir := t.TempDir()
+			archivePath := path.Join(archiveDir, "archive.tar")
+			{
+				// Create context to close the file handlers
+				archiveFile, err := os.Create(archivePath)
+				Require(t, err)
+				defer archiveFile.Close()
+				tarWriter := tar.NewWriter(archiveFile)
+				defer tarWriter.Close()
+				for _, relativePath := range testCase.archiveFiles {
+					filePath := path.Join(archiveDir, relativePath)
+					dir := filepath.Dir(filePath)
+					const dirPerm = 0700
+					err := os.MkdirAll(dir, dirPerm)
+					Require(t, err)
+					const filePerm = 0600
+					err = os.WriteFile(filePath, []byte{0xbe, 0xef}, filePerm)
+					Require(t, err)
+					file, err := os.Open(filePath)
+					Require(t, err)
+					info, err := file.Stat()
+					Require(t, err)
+					header, err := tar.FileInfoHeader(info, "")
+					Require(t, err)
+					header.Name = relativePath
+					err = tarWriter.WriteHeader(header)
+					Require(t, err)
+					_, err = io.Copy(tarWriter, file)
+					Require(t, err)
+				}
+			}
+
+			// Extract archive and compare contents
+			targetDir := t.TempDir()
+			err := extractSnapshot(archivePath, targetDir, testCase.importWasm)
+			Require(t, err, "failed to extract snapshot")
+			gotFiles := []string{}
+			err = filepath.WalkDir(targetDir, func(path string, d os.DirEntry, err error) error {
+				if err != nil {
+					return err
+				}
+				if !d.IsDir() {
+					gotFiles = append(gotFiles, path)
+				}
+				return nil
+			})
+			Require(t, err)
+			slices.Sort(gotFiles)
+			for i, f := range testCase.wantFiles {
+				testCase.wantFiles[i] = path.Join(targetDir, f)
+			}
+			if diff := cmp.Diff(gotFiles, testCase.wantFiles); diff != "" {
+				t.Fatal("extracted files don't match", diff)
+			}
+		})
+	}
+}
+
+func TestIsWasmDb(t *testing.T) {
+	testCases := []struct {
+		path string
+		want bool
+	}{
+		{"wasm", true},
+		{"wasm/", true},
+		{"wasm/something", true},
+		{"/wasm", true},
+		{"./wasm", true},
+		{"././wasm", true},
+		{"/./wasm", true},
+		{"WASM", true},
+		{"wAsM", true},
+		{"nitro/../wasm", true},
+		{"/nitro/../wasm", true},
+		{".//nitro/.//../wasm", true},
+		{"not-wasm", false},
+		{"l2chaindata/example@@", false},
+		{"somedir/wasm", false},
```

Comment:
Let's also test `wasm\0test` and `wasm\0test/example`, as I think on Linux this'll result in it being named just "wasm" as it's cstr based.

(I think I accidentally put the comment on the commit instead of the PR before, sorry if this is repeated)

Reply:
What about other formats, like zip? Alternatively, could you try what happens if you return `wasm\0test` in the rename function? That'd be a good test of what the rest of the extraction pipeline does with that. If that errors out, then I think it's fine

Reply:
I changed the function to check for the prefix, so we will also disallow anything else that starts with wasm.

Reply:
> What about other formats, like zip? Alternatively, could you try what happens if you return wasm\0test in the rename function? That'd be a good test of what the rest of the extraction pipeline does with that. If that errors out, then I think it's fine

If I add `return "foo\x00" + path` to the rename function, I get:
```
    init_test.go:656:  [failed to extract snapshot] couln't extract init archive '/var/folders/fl/7_c5xmcn727fhm_pyybvjfq00000gn/T/TestExtractSnapshotextractAllButWasmWithPrefixDot2916370482/001/archive.tar' err: Create file /var/folders/fl/7_c5xmcn727fhm_pyybvjfq00000gn/T/TestExtractSnapshotextractAllButWasmWithPrefixDot2916370482/002/foo./arbitrumdata/000001.ldb: mkdir /var/folders/fl/7_c5xmcn727fhm_pyybvjfq00000gn/T/TestExtractSnapshotextractAllButWasmWithPrefixDot2916370482/002/foo.: invalid argument
```

Reply:
Sounds good. No need to check for prefix then, as that fails as expected

Reply:
> Sounds good. No need to check for prefix then, as that fails as expected

Ok! I reverted the last commit.